### PR TITLE
ONME-3984 Refactor IP_Route for interface name Initial release

### PR DESCRIPTION
**Pull request by tymoteuszblochmobica**
### Description
Initial release.  
To do: set netif to PCB  for ICMP

### Pull request type

    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [x] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### CODE REVIEW

**Comment by tymoteuszblochmobica** [comment]
@mikaleppanen please review

**Comment by tymoteuszblochmobica** [comment]
@Seppotakalo please review

**Comment by tymoteuszblochmobica** [comment]
@kjbracey-arm please review

File: features/lwipstack/lwip/src/core/ipv4/lwip_icmp.c [hunk 1]
```diff
@@ -371,14 +371,18 @@ icmp_send_response(struct pbuf *p, u8_t type, u8_t code)
           IP_HLEN + ICMP_DEST_UNREACH_DATASIZE);
 
   ip4_addr_copy(iphdr_src, iphdr->src);
+
+  char name [INTERFACE_NAME_SIZE];
+  sprintf(name, "%c%c%d", p->netif->name[0], p->netif->name[1], p->netif->num);
```
Comment (mikaleppanen) [inline features/lwipstack/lwip/src/core/ipv4/lwip_icmp.c]: This is made in several places so I think there should be helper functions for this. Also the p->netif->num is running from 0 to 255 so more space should be reserved for it.

**Review by mikaleppanen — CHANGES_REQUESTED** [review]

**Pull request closed without merging**